src/studio/src/designer/backend/Services/Implementation/RepositorySI.cs

```diff
@@ -1163,6 +1168,70 @@ public async Task<RepositoryClient.Model.Repository> CreateService(string org, S
             return repository;
         }
 
+        /// <inheritdoc/>
+        public async Task<RepositoryClient.Model.Repository> CopyRepository(string org, string sourceRepository, string targetRepository, string developer)
+        {
+            var options = new RepositoryClient.Model.CreateRepoOption(targetRepository);
+
+            RepositoryClient.Model.Repository repository = await CreateRepository(org, options);
+
+            if (repository == null || repository.RepositoryCreatedStatus != System.Net.HttpStatusCode.Created)
+            {
+                return repository;
+            }
+
+            string targetRepositortyPath = _settings.GetServicePath(org, targetRepository, developer);
+
+            if (Directory.Exists(targetRepositortyPath))
+            {
+                // "Soft-delete" of local repo folder with same name to make room for clone of the new repo
+                string backupPath = _settings.GetServicePath(org, $"{targetRepository}_REPLACED_BY_NEW_CLONE_{DateTime.Now.Ticks}", developer);
+                Directory.Move(targetRepositortyPath, backupPath);
+            }
+
+            _sourceControl.CloneRemoteRepository(org, targetRepository);
+            var targetAppRepository = _altinnGitRepositoryFactory.GetAltinnAppGitRepository(org, targetRepository, developer);
+
+            // clone source repository
+            string sourceCloneName = $"{sourceRepository}_COPY_OF_ORIGIN_{DateTime.Now.Ticks}";     
+            _sourceControl.CloneRemoteRepository(org, sourceRepository, _settings.GetServicePath(org, sourceCloneName, developer));
+
+            var sourceAppRepository = _altinnGitRepositoryFactory.GetAltinnAppGitRepository(org, sourceCloneName, developer);
+
+            try
+            {
+                sourceAppRepository.CopyRepository(targetAppRepository.RepositoryDirectory);
+            }
+            finally
+            {
+                Directory.Delete(sourceAppRepository.RepositoryDirectory, true);
+            }
+
+            await UpdateRemoteOrigin(targetAppRepository, sourceRepository);
+            await UpdateAppId(targetAppRepository);
+
+            CommitInfo commitInfo = new CommitInfo() { Org = org, Repository = targetRepository, Message = $"App cloned from {sourceRepository} {DateTime.Now.Date.ToShortDateString()}" };
+            _sourceControl.PushChangesForRepository(commitInfo);
+
+            return repository;
+        }
+
+        private static async Task UpdateRemoteOrigin(AltinnGitRepository repository, string oldRepository)
```

Comment:
This method could be moved to the GitRepository class as it's pure git/file operations.

Reply:
The method uses information about the org, and has knowledge about how the remote path is build up, so maybe AltinnGitRepository instead

Reply:
Yes, you're right - good point:)

Reply:
Alternatively I guess we could send both search and replace string as input to the method and we would be fine.

Reply:
Yes, could do that as well - your call:)